armi/reactor/reactors.py

```diff
@@ -639,8 +646,8 @@ def removeAllAssemblies(self, discharge=True):
         assems = set(self)
         for a in assems:
             self.removeAssembly(a, discharge)
-        if hasattr(self.parent, "sfp"):
-            self.parent.sfp.removeAll()
+        if hasattr(self.parent, "excore") and self.parent.excore.get("sfp"):
+            self.parent.excore.sfp.removeAll()
```

Comment:
This check feels a little suspect...

If we need to check the SFP here, why do we not also need to check all the other ex-core structures for assemblies as well?

As a side note, I cannot find a single place that `removeAllAssemblies()` is used, and I can't really imagine what its for...

Reply:
Agreed. I'll just remove this...